mantidproject/mantid
# New LeanElasticPeaksWorkspace class

**Description of work.**

Follow on to #30812. Add the new Peak Workspace for LeanElasticPeak

**To test:**

```python
from mantid.simpleapi import *
from mantid.dataobjects import PeaksWorkspace, LeanElasticPeaksWorkspace

p=WorkspaceFactory.createPeaks('LeanElasticPeaksWorkspace')

print(isinstance(p, PeaksWorkspace))
print(isinstance(p, LeanElasticPeaksWorkspace))

p.addPeak(p.createPeakQSample(( 1, 0, 0)))
p.addPeak(p.createPeakQSample(( 0, 1, 0)))
p.addPeak(p.createPeakQSample(( 0, 0, 1)))
p.addPeak(p.createPeakQSample((-1, 0, 0)))
p.addPeak(p.createPeakQSample(( 0,-1, 0)))
p.addPeak(p.createPeakQSample(( 0, 0,-1)))

AnalysisDataService.addOrReplace('p',p)

SetUB(p)

p.addPeak(p.createPeakHKL((1,0,0)))
p.addPeak(p.createPeakHKL((0,-100,0)))
p.addPeak(p.createPeakHKL((-1,-1,-1)))

peak = p.createPeakHKL((1,1,1))
peak.setWavelength(1.5)
peak.setIntensity(100)
peak.setSigmaIntensity(10)
peak.setBinCount(50)
peak.setRunNumber(12345)
peak.setAbsorptionWeightedPathLength(1.1)
peak.setPeakNumber(99)
p.addPeak(peak)
```

After which you should be able to view the workspace in the workbench.


---

Updates for later PR's
* Once this is in a usable state all the documentation will be updated to reflect on how to use the new LeanElasticPeak.
* customize the columns for this _table workspace_, _i.e._ remove TOF, etc
* Add ability to save and load from nexus
* Make algorithms be able to use the LeanElasticPeaksWorkspace


---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
